os/binfmt/libelf/libelf_read.c

```diff
@@ -108,6 +108,46 @@ static inline void elf_dumpreaddata(FAR char *buffer, int buflen)
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: bin_read_header
+ *
+ * Description:
+ *   Read 'readsize' bytes from the object file.  The data is
+ *   read into 'buffer'.
+ *
+ * Returned Value:
+ *   0 (OK) is returned on success and a negated errno is returned on
+ *   failure.
+ *
+ ****************************************************************************/
+ int bin_read_header(int fd, FAR uint8_t *buffer, size_t readsize)
+ {
+	ssize_t nbytes;				/* Number of bytes read */
+
+	binfo("Read %ld bytes\n", (long)readsize);
+
+	/* Loop until all of the requested data has been read. */
+	while (readsize > 0) {
+
+		nbytes = read(fd, buffer, readsize);
+		if (nbytes < 0) {
+			/* EINTR just means that we received a signal */
+
+			if (nbytes != -EINTR) {
+				berr("Read failed: %d\n",(int)nbytes);
+				return nbytes;
+			}
+		} else if (nbytes == 0) {
+			berr("Unexpected end of file\n");
+			return -ENODATA;
+		} else {
+			readsize -= nbytes;
+			buffer += nbytes;
```

Comment:
readsize is value. can we modify?
buffer is pointer value. I think *buffer is correct.

Reply:
No issues with modifying the readsize.
buffer is correct we are increment the point not the content.